### PR TITLE
Add functionality for non-SSL endpoints for backwards compatibility

### PR DESCRIPTION
**Pull request by isaiahvita**
New endpoint resolution doesnt support non-SSL endpoints. But old endpoint resolution does. This middleware adds support for legacy non-SSL endpoints for backwards compatibility purposes.

### CODE REVIEW

**Review by lucix-aws — APPROVED** [review]

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/EndpointDisableHttps.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+ import software.amazon.smithy.go.codegen.SymbolUtils;
+ import software.amazon.smithy.go.codegen.endpoints.EndpointMiddlewareGenerator;
+ import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.MiddlewareRegistrar;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.Model;
+ import software.amazon.smithy.model.knowledge.TopDownIndex;
+ import software.amazon.smithy.model.shapes.OperationShape;
+ import software.amazon.smithy.model.shapes.ToShapeId;
+import software.amazon.smithy.utils.ListUtils;
```
Comment (syall) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/EndpointDisableHttps.java:38]: Weird spacing throughout the whole file.

File: service/s3/internal/customizations/update_endpoint_test.go [hunk 2]
```diff
@@ -35,33 +35,6 @@ func Test_UpdateEndpointBuild(t *testing.T) {
 		customEndpoint *aws.Endpoint
 	}{
 		"default endpoint": {
```
Comment (syall) [inline service/s3/internal/customizations/update_endpoint_test.go:37]: Removing these tests probably shouldn't be in this PR.

**Review by syall — CHANGES_REQUESTED** [review]

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/EndpointDisableHttps.java (hunk 1 above)
Comment (isaiahvita) [reply codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/EndpointDisableHttps.java]: fixed

File: service/s3/internal/customizations/update_endpoint_test.go (hunk 2 above)
Comment (isaiahvita) [reply service/s3/internal/customizations/update_endpoint_test.go]: fixed

**Review by syall — APPROVED** [review]

**Pull request merged**